source/src/cip/cipcommon.c

```diff
@@ -1264,6 +1289,146 @@ int DecodePaddedEPath(CipEpath *epath,
   return number_of_decoded_elements * 2 + 1; /* number_of_decoded_elements times 2 as every encoding uses 2 bytes */
 }
 
+EipStatus Create(CipInstance *RESTRICT const instance,
+                             CipMessageRouterRequest *const message_router_request,
+                             CipMessageRouterResponse *const message_router_response,
+                             const struct sockaddr *originator_address,
+                             const int encapsulation_session) {
+
+  InitializeENIPMessage(&message_router_response->message);
+  message_router_response->reply_service = (0x80 | message_router_request->service);
+  message_router_response->general_status = kCipErrorSuccess;
+  message_router_response->size_of_additional_status = 0;
+
+  CipClass *class = GetCipClass(message_router_request->request_path.class_id);
+
+  EipStatus internal_state = kEipStatusOk;
+
+  /* Call the PreCreateCallback if the class provides one. */
+  if( NULL != class->PreCreateCallback) {
+    internal_state = class->PreCreateCallback(instance, message_router_request, message_router_response);
+  }
+
+  if (kEipStatusOk == internal_state) {
+    CipInstance *new_instance = AddCipInstances(class, 1); /* add 1 instance to class*/
+    OPENER_ASSERT(NULL != new_instance); /* fail if run out of memory */
+
+    /* Call the PostCreateCallback if the class provides one. */
+    if (NULL != class->PostCreateCallback) {
+      class->PostCreateCallback(new_instance, message_router_request, message_router_response);
+    }
+    OPENER_TRACE_INFO("Instance number %d created\n", new_instance->instance_number);
+  }
+  return kEipStatusOkSend;
+}
+
+EipStatus Delete(CipInstance *RESTRICT const instance,
```

Comment:
Delete is probably a too generic name, please change to CipDeleteService (or similar)